packages/repocop/src/metrics.ts

```diff
@@ -0,0 +1,86 @@
+import {
+	type CloudWatchClient,
+	type MetricDatum,
+	PutMetricDataCommand,
+} from '@aws-sdk/client-cloudwatch';
+import type { repocop_github_repository_rules } from '@prisma/client';
+import { getEnvOrThrow } from 'common/src/functions';
+
+export function getPercentageTrue(booleanishArray: Array<boolean | null>) {
+	const array = booleanishArray.filter((x) => x !== null) as boolean[];
+	if (array.length === 0) {
+		return 0;
+	}
+	const totalRepos = array.length;
+	const trackedReposPercentage = (
+		(array.filter((x) => x).length * 100) /
+		totalRepos
+	).toFixed(1);
+	return parseFloat(trackedReposPercentage);
+}
+
+function createMetric(
+	metricName: string,
+	boolArray: Array<boolean | null>,
+): MetricDatum {
+	getPercentageTrue(boolArray);
+	const Dimensions = [
+		{
+			Name: 'Stack',
+			Value: getEnvOrThrow('STACK'),
+		},
+		{
+			Name: 'Stage',
+			Value: process.env.STAGE ?? 'DEV',
+		},
+		{
+			Name: 'App',
+			Value: getEnvOrThrow('APP'),
+		},
+	];
+
+	return {
+		MetricName: metricName,
+		Value: getPercentageTrue(boolArray),
+		Unit: 'Percent',
+		Dimensions,
+	};
+}
+
+export async function sendToCloudwatch(
+	evaluatedRepos: repocop_github_repository_rules[],
+	cloudwatch: CloudWatchClient,
+) {
+	console.log('Sending metrics to Cloudwatch');
+	await cloudwatch.send(
+		new PutMetricDataCommand({
+			Namespace: 'repocop',
+			MetricData: [
+				createMetric(
+					'DefaultBranchIsMain',
+					evaluatedRepos.map((x) => x.default_branch_name),
```

Comment:
Again, up to you, but you could destructure these to avoid defining a parameter `x`.

```suggestion
					evaluatedRepos.map(({ default_branch_name }) => default_branch_name),
```

Reply:
I tried this suggestion, but got a linting error so will leave it how it is for now

<img width="895" alt="Screenshot 2024-01-16 at 13 55 46" src="https://github.com/guardian/service-catalogue/assets/67543397/d7697326-625d-4ee3-9f82-6f0e98318bf4">

Reply:
I tried this and got a bunch of linting errors, so will leave it how it is for now